packages/yew-macro/src/props/component.rs

```diff
@@ -9,6 +9,27 @@ use syn::{
     Expr, ExprLit, Lit,
 };
 
+struct BaseExpr {
+    pub dot2: Dot2,
+    pub expr: Expr,
+}
+
+impl Parse for BaseExpr {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let dot2 = input.parse()?;
+        let expr = input.parse()?;
```

Comment:
```suggestion
        let expr = input.parse().map_err(|_| {
            syn::Error::new_spanned(dot2, "props base required after `..`")
        })?;
```
Rust provides a very similar error for structs and with this new type we can do it neatly here too :)

Before:
```bash
error: unexpected end of input, expected expression
  --> examples/counter/src/main.rs:41:17
   |
41 |                 <Comp .. />
   |                 ^^^^^^^^^^^
```

After:
```bash
error: props base required after `..`
  --> examples/counter/src/main.rs:45:34
   |
45 |                 <Comp prop1="hi" .. />
   |                                  ^^

```
Think this is more clear after. 

_Note: This will break the trybuild test so will have to update `component-fail.stderr` too_

Reply:
Instead of ignoring the error from attempting to parse the expression, I combined them so that the error that the user sees is more descriptive

Reply:
Very nice :D